scripts/check_style.sh

```diff
@@ -3,54 +3,54 @@
 ERROR_LOG="$(mktemp -t check_style_XXXXXX.log)"
 
 EXCLUDE_FILES=(
-  "libsolutil/picosha2.h"
-  "test/libsolutil/UTF8.cpp"
+    "libsolutil/picosha2.h"
+    "test/libsolutil/UTF8.cpp"
 )
 EXCLUDE_FILES_JOINED=$(printf "%s\|" "${EXCLUDE_FILES[@]}")
 EXCLUDE_FILES_JOINED=${EXCLUDE_FILES_JOINED%??}
 
 (
 REPO_ROOT="$(dirname "$0")"/..
-cd $REPO_ROOT
+cd "$REPO_ROOT" || exit 1
 
 WHITESPACE=$(git grep -n -I -E "^.*[[:space:]]+$" |
-  grep -v "test/libsolidity/ASTJSON\|test/libsolidity/ASTRecoveryTests\|test/compilationTests/zeppelin/LICENSE\|${EXCLUDE_FILES_JOINED}"
+    grep -v "test/libsolidity/ASTJSON\|test/libsolidity/ASTRecoveryTests\|test/compilationTests/zeppelin/LICENSE\|${EXCLUDE_FILES_JOINED}"
 )
 
 if [[ "$WHITESPACE" != "" ]]
 then
-	echo "Error: Trailing whitespace found:" | tee -a $ERROR_LOG
-	echo "$WHITESPACE" | tee -a $ERROR_LOG
-	scripts/report_errors.sh "$ERROR_LOG"
-	exit 1
+    echo "Error: Trailing whitespace found:" | tee -a "$ERROR_LOG"
+    echo "$WHITESPACE" | tee -a "$ERROR_LOG"
+    scripts/post_style_errors_on_github.sh "$ERROR_LOG"
+    exit 1
 fi
 
 function preparedGrep()
 {
-	git grep -nIE "$1" -- '*.h' '*.cpp' | grep -v "${EXCLUDE_FILES_JOINED}"
-	return $?
+    git grep -nIE "$1" -- '*.h' '*.cpp' | grep -v "${EXCLUDE_FILES_JOINED}"
+    return $?
 }
 
 FORMATERROR=$(
 (
-	preparedGrep "#include \"" | egrep -v -e "license.h" -e "BuildInfo.h"  # Use include with <> characters
-	preparedGrep "\<(if|for|while|switch)\(" # no space after "if", "for", "while" or "switch"
-	preparedGrep "\<for\>\s*\([^=]*\>\s:\s.*\)" # no space before range based for-loop
-	preparedGrep "\<if\>\s*\(.*\)\s*\{\s*$" # "{\n" on same line as "if" / "for"
```

Comment:
The comment wasn't true so I removed the part about `for`. But why actually aren't we checking `for`? Is there any reason this isn't `(if|for|while|switch)`? I checked blame and it looks like it was `if` from the beginning.

Reply:
I personally have no idea why.
@chriseth @axic ?

Reply:
Please add all of them.

Reply:
OK. But in that case I'll do it in a separate PR because this will require fixing the style issues that were passing though this check until now.